src/NotesApplication/Models/Note.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+namespace NotesApplication
+{
+    ///<summary>
+    /// Represents a note in the database.
+    ///
+    /// This class provides methods that allow easy manipulation of a Note in the database.
+    /// Each Note object that is an instance of this class represents a physical note stored in the database.
+    /// Because of this, Note objects <b>must</b> be obtained from the static functions in this class, (FromDatabase, and Create). This helps assure that each created C# Note object represents a note in the database.
+    ///</summary>
+    public class Note
+    {
+        /// <summary>
+        /// Unique ID of the note
+        /// </summary>
+        internal string id { get; } //We should not allow updating of IDs because it could mess with the database
+
+        /// <summary>
+        /// The name of the note. (Similar to a title)
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The files attached to the note
+        /// </summary>
+        public string[] Notes { get; set; } //Wasn't this suppose to be a HashMap?
```

Comment:
nit: Maybe add a bit more documentation explaining that this is a location/uri for the notes n Azure Blob Store. At first glance, I can understand this as just the ID of the notes or really anything else